Compiler/doc/languageGrammar.txt

```diff
@@ -0,0 +1,60 @@
+Grammar
+
+Here is the grammar for Klein. In this grammar, e, the lowercase Greek letter epsilon, stands for the empty string. It indicates that nothing is a legal alternative.
+
+            <PROGRAM> ::= <DEFINITIONS>
+
+        <DEFINITIONS> ::= e
+                        | <DEF> <DEFINITIONS>
+
+                <DEF> ::= function <IDENTIFIER> ( <FORMALS> ) : <TYPE>
+                             <BODY>
+
+            <FORMALS> ::= e
+                        | <NONEMPTYFORMALS>
+
+    <NONEMPTYFORMALS> ::= <FORMAL>
+                        | <FORMAL> , <NONEMPTYFORMALS>
+
+             <FORMAL> ::= <IDENTIFIER> : <TYPE>
+
+               <BODY> ::= <PRINT-STATEMENT> <BODY>
+                        | <EXPR>
+
+               <TYPE> ::= integer
+                        | boolean
+
+               <EXPR> ::= <EXPR> < <SIMPLE-EXPR>
+                        | <EXPR> = <SIMPLE-EXPR>
+                        | <SIMPLE-EXPR>
+
+        <SIMPLE-EXPR> ::= <SIMPLE-EXPR> or <TERM>
+                        | <SIMPLE-EXPR> + <TERM>
+                        | <SIMPLE-EXPR> - <TERM>
+                        | <TERM>
+
+               <TERM> ::= <TERM> and <FACTOR>
+                        | <TERM> * <FACTOR>
+                        | <TERM> / <FACTOR>
+                        | <FACTOR>
+
+             <FACTOR> ::= if <EXPR> then <EXPR> else <EXPR>
+                        | not <FACTOR>
+                        | <IDENTIFIER> ( <ACTUALS> )
+                        | <IDENTIFIER>
+                        | <LITERAL>
+                        | - <FACTOR>
+                        | ( <EXPR> )
+
+            <ACTUALS> ::= e
+                        | <NONEMPTYACTUALS>
+
+    <NONEMPTYACTUALS> ::= <EXPR>
```

Comment:
This NONEMPTYACTUALS has  <EXP> on both sides of the | so it's ambiguous still. There's quite a few places this happens yet (SIMPLE-EXPR, TERM, NONEMPTYFORMALS)  So we need to fix that.